### ISSUE TITLE
Jira 6 support

### ISSUE THREAD

**Issue by jeanarcand** [jeanarcand/foresight#14]
Might already work, setup an instance and see how it goes.

**Comment by perolo** [comment]
Try using the REST API - I believe you will have less problems in the future with new versions of JIRA?

**Comment by jeanarcand** [comment]
Yes some parts will need a rewrite to properly make use of jira's api's. I'll have to find a balance between supporting jira 4.x and using the latest and greatest from the api's to be future-proof.

**Comment by jawspeak** [comment]
Do you have any list of work that is required to work for jira 6? i'd be interested in helping.

**Comment by jeanarcand** [comment]
Hey, nop I don't, my plan was to simply setup a jira 6 instance and work from there. 

I'd probably start by generating a version 6 plugin using the "atlas-create-jira-plugin" command of the sdk. Compare the newly generated pom.xml and the existing one from the master branch. If you update the pom.xml, some changes are probably gonna be needed here and there if they changed some api's (IssueManager, IssueLinkManager, etc.).

Some changes might also be required in atlassian-plugin.xml but that'd be surprising since the plugin is only using basic hooks (issue tab panel and some servlets).

**Comment by jawspeak** [comment]
Actually we tried today. Appears it works!

It'd be awesome to have a filter create the stories to include in the
graph, and then edges be drawn. But that's a feature request, not an issue.

On Wednesday, November 6, 2013, Jean Arcand wrote:

> Hey, nop I don't, my plan was to simply setup a jira 6 instance and work
> from there.
> 
> I'd probably start by generating a version 6 plugin using the
> "atlas-create-jira-plugin" command of the sdk. Compare the newly generated
> pom.xml and the existing one from the master branch. If you update the
> pom.xml, some changes are probably gonna be needed here and there if they
> changed some api's (IssueManager, IssueLinkManager, etc.).
> 
> Some changes might also be required in atlassian-plugin.xml but that'd be
> surprising since the plugin is only using basic hooks (issue tab panel and
> some servlets).
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jeanarcand/foresight/issues/14#issuecomment-27933499
> .

## 

Sent from my Phone

**Comment by jonathanvaughn** [comment]
Using that small hint about pom.xml, I installed the atlassian SDK tools and created a new plugin, then diff'd the pom.xml and copied values as seemed appropriate. I had to modify one line of src/main/java/ca/appbox/jira/plugins/issuedependencyviewer/issuetabpanels/IssueDependencyTabPanel.java to recompile it though:

(both diffs are new / modified <, original github pulled source > )

``` diff
12c12
< import com.atlassian.crowd.embedded.api.User;
---
> import com.opensymphony.user.User;
```

The changes made based on the new pom.xml which appears to work for us (this may not be optimal, but it works)

``` diff
24,32c24
<             <artifactId>jira-api</artifactId>
<             <version>${jira.version}</version>
<             <scope>provided</scope>
<         </dependency>
<         <!-- Add dependency on jira-core if you want access to JIRA implementation classes as well as the sanctioned API. -->
<         <!-- This is not normally recommended, but may be required eg when migrating a plugin originally developed against JIRA 4.x -->
<         <dependency>
<             <groupId>com.atlassian.jira</groupId>
<             <artifactId>jira-core</artifactId>
---
>             <artifactId>atlassian-jira</artifactId>
57,61d48
<             <groupId>com.google.code.gson</groupId>
<             <artifactId>gson</artifactId>
<             <version>2.2.2-atlassian-1</version>
<         </dependency>
<         <dependency>
79c65
<                     <productDataVersion>${jira.version}</productDataVersion>
---
>                     <productDataVersion>${jira.data.version}</productDataVersion>
93,95c79,82
<         <jira.version>6.1.3</jira.version>
<         <amps.version>4.2.10</amps.version>
<         <plugin.testrunner.version>1.1.2</plugin.testrunner.version>
---
>         <jira.version>4.1.1</jira.version>
>         <jira.data.version>4.4</jira.data.version>
>         <amps.version>4.1.6</amps.version>
>         <plugin.testrunner.version>1.1.1</plugin.testrunner.version>
```

I didn't try using older versions for compatibility with older JIRA (if that is even how it's determined??), since we're not running it and I didn't want to waste time rebuilding it again and again if it didn't work trying to find the magic combination.

Not bad for never having touched the SDK before today ... :D

**Comment by peterjenkins** [comment]
Would love to try this on my JIRA 6.1.7 instance. It doesn't show up in the marketplace.

Any plans to make it so it can be installed from the Add On administration page automatically?

Not really sure what's involved in doing so.

**Comment by Beartogan** [comment]
Is their any chance, that the Jira 6 compatible version will be available on the Jira marketplace?

**Comment by peterjenkins** [comment]
@Beartogan I don't wish to take away from what I'm sure is an excellent plug-in (foresight). And admirably open source. 

But, wanted to share that since I asked about 6.1.7 support, I found [Links Hierarchy](https://marketplace.atlassian.com/plugins/com.docminer.jira.issue-links/versions#b711) which is paid but extremely nice. And I think the price is pretty reasonable. It's definitely the most functional out of a few different link hierarchy plugins that I tried.

Here's a screenshot: https://marketplace-cdn.atlassian.com/files/images/com.docminer.jira.issue-links/c2fabc5c-37a2-440c-af53-3d595608740b.png

@jeanarcand My apologies for promoting a competing product on your repo. I'm not associated with it, if it helps. Still would be interested in trying yours out if it pops up on the marketplace.

**Comment by jeanarcand** [comment]
Did some quick testing against jira 6.2 with the latest foresight and so far so good, just seems to work.

As for making it available in the marketplace for jira 6+, I think it's just a matter of releasing a new version and setting the compatibility info accordingly.

@pjenkins-cc No worries, I already came across Links Hierarchy which seems to be an awesome product but unfortunately wasn't compatible with jira 4 which was the version my former client used. Which is the reason I developped foresight. Still, there's always a place for a free alternative!
